Core/Graphics/Source/Graphics.cpp

```diff
@@ -0,0 +1,191 @@
+        }
+
+        if (workDone)
+        {
+            bgfx::frame();
+        }
+
+        auto oldRenderTaskCompletionSource = AfterRenderTaskCompletionSource;
+        AfterRenderTaskCompletionSource = {};
+        oldRenderTaskCompletionSource.complete();
+
+        m_rendering = false;
+    }
+
+    arcana::task<void, std::exception_ptr> Graphics::Impl::RenderCurrentFrameAsync(bool& finished, bool& workDone)
+    {
+        bool anyTasks{};
+        arcana::task<void, std::exception_ptr> whenAllTask{};
+        {
+            std::scoped_lock RenderWorkTasksLock{RenderWorkTasksMutex};
+            anyTasks = !RenderWorkTasks.empty();
+            if (anyTasks)
+            {
+                whenAllTask = arcana::when_all<std::exception_ptr>(RenderWorkTasks);
+                RenderWorkTasks.clear();
+            }
+        }
+
+        workDone = workDone || anyTasks;
+
+        if (!anyTasks)
+        {
+            finished = true;
+            return arcana::task_from_result<std::exception_ptr>();
+        }
+        else
+        {
+            return whenAllTask.then(Dispatcher, arcana::cancellation::none(), [this, &finished, &workDone]() mutable {
+                return RenderCurrentFrameAsync(finished, workDone);
+            });
+        }
+    }
+
+    Graphics::Graphics()
+        : m_impl{std::make_unique<Graphics::Impl>()}
+    {
+    }
+
+    Graphics::~Graphics() = default;
+
+    template<>
+    std::unique_ptr<Graphics> Graphics::InitializeFromWindow<void*>(void* nativeWindowPtr, size_t width, size_t height)
+    {
+        std::unique_ptr<Graphics> graphics{new Graphics()};
+
+        // Initialize bgfx.
+        bgfx::Init init{};
+        init.platformData.nwh = nativeWindowPtr;
+        bgfx::setPlatformData(init.platformData);
+#if (ANDROID)
+        init.type = bgfx::RendererType::OpenGLES;
+#else
+        init.type = bgfx::RendererType::Direct3D11;
+#endif
+        init.resolution.width = static_cast<uint32_t>(width);
+        init.resolution.height = static_cast<uint32_t>(height);
+        init.resolution.reset = BGFX_RESET_FLAGS;
+        init.callback = &graphics->m_impl->Callback;
+        bgfx::init(init);
+        bgfx::setViewClear(0, BGFX_CLEAR_COLOR | BGFX_CLEAR_DEPTH, 0x443355FF, 1.0f, 0);
+        bgfx::setViewRect(0, 0, 0, static_cast<uint16_t>(init.resolution.width), static_cast<uint16_t>(init.resolution.height));
+        bgfx::touch(0);
+
+        return graphics;
+    }
+
+    template<>
+    void Graphics::ReinitializeFromWindow<void*>(void* windowPtr, size_t width, size_t height)
+    {
+        bgfx::PlatformData pd{};
+        pd.ndt = nullptr;
+        pd.nwh = windowPtr;
+        pd.context = nullptr;
+        pd.backBuffer = nullptr;
+        pd.backBufferDS = nullptr;
+        bgfx::setPlatformData(pd);
+        bgfx::reset(static_cast<uint32_t>(width), static_cast<uint32_t>(height));
+    }
+
+    void Graphics::Impl::AddToJavaScript(Napi::Env env)
```

Comment:
Kind of. The Graphics component is also designed to support multi-`Env` scenarios, at least in theory; and though we're currently resetting the Graphics component a lot to match existing behavior, long-term the usage pattern will more likely tend (I think) toward creating the Graphics once at the beginning, after which it will survive an arbitrary number of Runtimes.